scalyr_agent/builtin_monitors/tests/kubernetes_monitor_test.py

```diff
@@ -555,3 +567,145 @@ def run_test():
+                ),
+                InsecureRequestWarning,
+            )
+            return resp
+
+        with mock.patch.object(KubeletApi, "_get", fake_get):
+            new_out, new_err = StringIO(), StringIO()
+            old_out, old_err = sys.stdout, sys.stderr
+            try:
+                sys.stdout, sys.stderr = new_out, new_err
+                api = KubeletApi(
+                    FakeKubernetesApi(),
+                    host_ip="127.0.0.1",
+                    node_name="FakeNode",
+                    kubelet_url_template=Template("https://${host_ip}"),
+                    verify_https=False,
+                )
+                result = api.query_stats()
+                self.assertEqual(result, {})
+                self.assertLogFileContainsLineRegex(
+                    ".*"
+                    + re.escape(
+                        "WARNING [core] [k8s.py:2215] Accessing Kubelet with an unverified HTTPS request."
+                    )
+                )
+                self.assertFalse(new_err.getvalue())
+                self.assertFalse(new_out.getvalue())
+            finally:
+                sys.stdout, sys.stderr = old_out, old_err
+
+    def test_unverified_http(self):
+        def fake_get(self, url, verify):
+            resp = FakeResponse()
+            return resp
+
+        with mock.patch.object(KubeletApi, "_get", fake_get):
+            new_out, new_err = StringIO(), StringIO()
+            old_out, old_err = sys.stdout, sys.stderr
+            try:
+                sys.stdout, sys.stderr = new_out, new_err
+                api = KubeletApi(
+                    FakeKubernetesApi(),
+                    host_ip="127.0.0.1",
+                    node_name="FakeNode",
+                    kubelet_url_template=Template("http://${host_ip}"),
+                    verify_https=False,
+                )
+                result = api.query_stats()
+                self.assertEqual(result, {})
+                self.assertLogFileDoesntContainsLineRegex(
+                    ".*"
+                    + re.escape(
+                        "WARNING [core] [k8s.py:2215] Accessing Kubelet with an unverified HTTPS request."
+                    )
+                )
+                self.assertFalse(new_err.getvalue())
+                self.assertFalse(new_out.getvalue())
+            finally:
+                sys.stdout, sys.stderr = old_out, old_err
+
+    def test_error_response(self):
+        def fake_get(self, url, verify):
+            resp = FakeResponse()
+            resp.status_code = 404
+            return resp
+
+        with mock.patch.object(KubeletApi, "_get", fake_get):
+            try:
+                api = KubeletApi(
+                    FakeKubernetesApi(), host_ip="127.0.0.1", node_name="FakeNode",
+                )
+                api.query_stats()
+                self.assertTrue(False)
+            except KubeletApiException:
+                pass
+
```

Comment:
Can you also add in a test case where you return a `403` to trigger switching to the fallback URL?